Resources/AzureDevOps/azure-pipelines.yml

```diff
@@ -7,27 +9,113 @@ resources:
     type: github
     name: SkillsFundingAgency/dfc-devops
     endpoint: 'GitHub (ESFA)'
-    ref: refs/tags/v1.2.1
+    ref: refs/tags/v1.7.2
 
 pr:
   branches:
     include:
     - master
 
-steps:
+stages:
 
-# ARM template
-- template: AzureDevOpsTemplates/Build/dfc-arm-build.yml@dfc-devops
-  parameters:
-    ArmTemplateRoot: '$(System.DefaultWorkingDirectory)\Resources'
+- stage: Build
+  jobs:
+  - job: TestAndPublish
+    pool:
+      name: 'NCS - CI and CD'
+    variables:
+      - group: 'KeyVault - dfc-dev-shared-kv'
+      - group: dfc-shared-infrastructure-dev
     steps:
-- task: CopyFiles@2
-  displayName: 'Copy PSScripts Files to: $(Build.ArtifactStagingDirectory)'
-  inputs:
-    Contents: 'PSScripts/**/*.ps1'
-    TargetFolder: '$(Build.ArtifactStagingDirectory)'
-- task: PublishBuildArtifacts@1
-  displayName: Publish Artifact
-  inputs:
-     pathtoPublish: $(Build.ArtifactStagingDirectory)
-     artifactName: drop
\ No newline at end of file
+    # ARM template
+    - task: TotalALM.totalalm-tokenization.tokenization-build-task.Tokenization@2
+      displayName: 'Tokenization: Transform file parameters.json'
+      inputs:
+        SourcePath: '$(System.DefaultWorkingDirectory)\Resources'
+        TargetFileNames: test-parameters.json
+    - template: AzureDevOpsTemplates/Build/StepTemplates/dfc-arm-build.yml@dfc-devops
+      parameters:
+        ArmTemplateRoot: '$(System.DefaultWorkingDirectory)\Resources'
+        SolutionBaseName: $(SolutionBaseName)
+        PublishPipelineArtifact: true
+    - task: CopyFiles@2
+      displayName: 'Copy PSScripts Files to: $(Build.ArtifactStagingDirectory)'
+      inputs:
+        Contents: 'PSScripts/**/*.ps1'
+        TargetFolder: '$(Build.ArtifactStagingDirectory)/Resources/PSScripts'
+        flattenFolders: true
+    - task: PublishPipelineArtifact@0
+      displayName: Publish Pipeline Artifact
+      inputs:
+        targetPath: $(Build.ArtifactStagingDirectory)/Resources/PSScripts
+        artifactName: Dfc.Shared.Resources.PSScripts
+    # Also publish build artifact for backwards compatibility with classic Release Definition

```

Comment:
FYI:  "classic" release definitions can see and use pipeline artifacts, so this really isn't needed

Reply:
Agreed, I will leave this comment open as a reminder